File: Quokka.RISCV.Integration.Tests/Client/Blinker/Source/BasicTest.cpp
include <stdint>
include <stdbool>
namespace UnitTests
{
	static class BasicTest
	{
		public: static void _main()
		{
			int counter = 0;
			double tmp = 0.5;
			while((counter < 1024))
			{
				counter = (counter + 1);
				tmp = (tmp * 2);
			}
		}
	}
}

File: Quokka.RISCV.Integration.Tests/CSharp2CPPTranslatorTests/Source/BasicTest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace UnitTests
{
    public static class BasicTest
    {
        public static int increment(int value)
        {
            return value + 1;
        }

        public static void incrementByRef(ref int value)
        {
            value++;
        }

        public static void EntryPoint()
        {
            int counter = 0;

            while(counter < 1024)
            {
                counter = increment(counter);
                incrementByRef(ref counter);
            }
        }
    }
}
